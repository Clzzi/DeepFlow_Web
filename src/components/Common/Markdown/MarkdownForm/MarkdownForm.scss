.MarkdownForm {
	.custom-html-style {
		div,
		p,
		code {
			font-size: var(--font-contents);
			white-space: pre-wrap;
			padding: 2px 5px;
		}

		img {
			margin: 0.5rem 0;
		}

		h2 {
			margin-bottom: 0.23rem !important;
		}

		blockquote {
			background-color: var(--bg-codeBlock);
			color: var(--black);
			border-left: 4px solid var(--main);
			border-radius: 5px;
			padding: 10px;
			margin: 2rem 0;
		}
	}

	.rc-md-editor .editor-container .sec-md .input {
		font-size: var(--font-contents);
	}

	.custom-html-style code,
	.custom-html-style pre {
		background-color: var(--lighterGray);
		border-radius: 5px;
	}

	code[class^='language'],
	code[class*='language'] {
		padding: 0 !important;
	}

	img {
		width: 100%;
		max-height: 450px;
		margin: auto;
		object-fit: cover;
	}
}
