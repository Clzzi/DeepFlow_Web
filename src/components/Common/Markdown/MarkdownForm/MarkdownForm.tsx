import { memo, useCallback } from 'react';
import classNames from 'classnames';
import { ClassNamesFn } from 'classnames/types';
import MdEditor from 'react-markdown-editor-lite';
import imageUpload from 'util/imageUpload';
import 'react-markdown-editor-lite/lib/index.css';

const style = require('./MarkdownForm.scss');
const cx: ClassNamesFn = classNames.bind(style);

export interface MarkdownFormProps {
	height?: string;
	maxHeight?: string;
	contents: string;
	onChangeContents: (text: string) => void;
	onChangeIsFocus?: () => void;
}

const MarkdownForm = ({
	height = '75vh',
	maxHeight = '75vh',
	contents,
	onChangeContents,
	onChangeIsFocus,
}: MarkdownFormProps): JSX.Element => {
	const handleImageUpload = useCallback((file: File): Promise<string> => {
		return new Promise((resolve) => {
			const reader: FileReader = new FileReader();
			reader.onload = async () => {
				await imageUpload(file)
				.then((response: string) => {
					resolve(response);
				});
			};
			
			reader.readAsDataURL(file);
		});
	}, []);

	return (
		<div className={cx('MarkdownForm')}>
			<MdEditor
				name='contents'
				value={contents}
				onChange={({ text }) => onChangeContents(text)}
				style={{ height, maxHeight }}
				syncScrollMode={[]}
				renderHTML={(text: string) => ''}
				placeholder="내용을 입력하세요..."
				onImageUpload={handleImageUpload}
				onFocus={onChangeIsFocus}
				onBlur={onChangeIsFocus}
			/>
		</div>
	);
};

export default memo(MarkdownForm);
