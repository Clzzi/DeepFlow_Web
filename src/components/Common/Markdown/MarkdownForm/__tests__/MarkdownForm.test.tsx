import { fireEvent, render } from '@testing-library/react';
import MarkdownForm, { MarkdownFormProps } from '../MarkdownForm';

describe('MarkdownForm', () => {
  const initialSetup = (props: Partial<MarkdownFormProps> = {}) => {
    const initialProps: MarkdownFormProps = {
      contents: '',
      onChangeContents: (text: string) => {},
      onChangeIsFocus: () => {},
      ...props,
    };

    const utils = render(<MarkdownForm {...initialProps} />);
    const input = utils.getByPlaceholderText('내용을 입력하세요...');

    return {
      ...utils,
      input,
    };
  };

  it('render correctly', () => {
    const { container } = initialSetup();
    expect(container).toMatchSnapshot();
  });

  it('well handling change', () => {
    const { input } = initialSetup();
    fireEvent.change(input);
  });
});