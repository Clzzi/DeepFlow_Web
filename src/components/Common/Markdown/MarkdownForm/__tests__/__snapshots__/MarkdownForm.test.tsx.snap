// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownForm render correctly 1`] = `
<div>
  <div
    class="MarkdownForm"
  >
    <div
      class="rc-md-editor "
      style="height: 75vh; max-height: 75vh;"
    >
      <div
        class="rc-md-navigation visible"
      >
        <div
          class="navigation-nav left"
        >
          <div
            class="button-wrap"
          >
            <span
              class="button button-type-header"
              title="Header"
            >
              <i
                class="rmel-iconfont rmel-icon-font-size"
              />
              <div
                class="drop-wrap hidden"
              >
                <ul
                  class="header-list"
                >
                  <li
                    class="list-item"
                  >
                    <h1>
                      H1
                    </h1>
                  </li>
                  <li
                    class="list-item"
                  >
                    <h2>
                      H2
                    </h2>
                  </li>
                  <li
                    class="list-item"
                  >
                    <h3>
                      H3
                    </h3>
                  </li>
                  <li
                    class="list-item"
                  >
                    <h4>
                      H4
                    </h4>
                  </li>
                  <li
                    class="list-item"
                  >
                    <h5>
                      H5
                    </h5>
                  </li>
                  <li
                    class="list-item"
                  >
                    <h6>
                      H6
                    </h6>
                  </li>
                </ul>
              </div>
            </span>
            <span
              class="button button-type-bold"
              title="Bold"
            >
              <i
                class="rmel-iconfont rmel-icon-bold"
              />
            </span>
            <span
              class="button button-type-italic"
              title="Italic"
            >
              <i
                class="rmel-iconfont rmel-icon-italic"
              />
            </span>
            <span
              class="button button-type-underline"
              title="Underline"
            >
              <i
                class="rmel-iconfont rmel-icon-underline"
              />
            </span>
            <span
              class="button button-type-strikethrough"
              title="Strikethrough"
            >
              <i
                class="rmel-iconfont rmel-icon-strikethrough"
              />
            </span>
            <span
              class="button button-type-unordered"
              title="Unordered list"
            >
              <i
                class="rmel-iconfont rmel-icon-list-unordered"
              />
            </span>
            <span
              class="button button-type-ordered"
              title="Ordered list"
            >
              <i
                class="rmel-iconfont rmel-icon-list-ordered"
              />
            </span>
            <span
              class="button button-type-quote"
              title="Quote"
            >
              <i
                class="rmel-iconfont rmel-icon-quote"
              />
            </span>
            <span
              class="button button-type-wrap"
              title="Line break"
            >
              <i
                class="rmel-iconfont rmel-icon-wrap"
              />
            </span>
            <span
              class="button button-type-code-inline"
              title="Inline code"
            >
              <i
                class="rmel-iconfont rmel-icon-code"
              />
            </span>
            <span
              class="button button-type-code-block"
              title="Code"
            >
              <i
                class="rmel-iconfont rmel-icon-code-block"
              />
            </span>
            <span
              class="button button-type-table"
              title="Table"
            >
              <i
                class="rmel-iconfont rmel-icon-grid"
              />
              <div
                class="drop-wrap hidden"
              >
                <ul
                  class="table-list wrap"
                  style="width: 135px; height: 89px;"
                >
                  <li
                    class="list-item "
                    style="top: 0px; left: 0px;"
                  />
                  <li
                    class="list-item "
                    style="top: 0px; left: 23px;"
                  />
                  <li
                    class="list-item "
                    style="top: 0px; left: 46px;"
                  />
                  <li
                    class="list-item "
                    style="top: 0px; left: 69px;"
                  />
                  <li
                    class="list-item "
                    style="top: 0px; left: 92px;"
                  />
                  <li
                    class="list-item "
                    style="top: 0px; left: 115px;"
                  />
                  <li
                    class="list-item "
                    style="top: 23px; left: 0px;"
                  />
                  <li
                    class="list-item "
                    style="top: 23px; left: 23px;"
                  />
                  <li
                    class="list-item "
                    style="top: 23px; left: 46px;"
                  />
                  <li
                    class="list-item "
                    style="top: 23px; left: 69px;"
                  />
                  <li
                    class="list-item "
                    style="top: 23px; left: 92px;"
                  />
                  <li
                    class="list-item "
                    style="top: 23px; left: 115px;"
                  />
                  <li
                    class="list-item "
                    style="top: 46px; left: 0px;"
                  />
                  <li
                    class="list-item "
                    style="top: 46px; left: 23px;"
                  />
                  <li
                    class="list-item "
                    style="top: 46px; left: 46px;"
                  />
                  <li
                    class="list-item "
                    style="top: 46px; left: 69px;"
                  />
                  <li
                    class="list-item "
                    style="top: 46px; left: 92px;"
                  />
                  <li
                    class="list-item "
                    style="top: 46px; left: 115px;"
                  />
                  <li
                    class="list-item "
                    style="top: 69px; left: 0px;"
                  />
                  <li
                    class="list-item "
                    style="top: 69px; left: 23px;"
                  />
                  <li
                    class="list-item "
                    style="top: 69px; left: 46px;"
                  />
                  <li
                    class="list-item "
                    style="top: 69px; left: 69px;"
                  />
                  <li
                    class="list-item "
                    style="top: 69px; left: 92px;"
                  />
                  <li
                    class="list-item "
                    style="top: 69px; left: 115px;"
                  />
                </ul>
              </div>
            </span>
            <span
              class="button button-type-image"
              style="position: relative;"
              title="Image"
            >
              <i
                class="rmel-iconfont rmel-icon-image"
              />
              <input
                accept=""
                style="position: absolute; z-index: -1; left: 0px; top: 0px; width: 0px; height: 0px; opacity: 0;"
                type="file"
              />
            </span>
            <span
              class="button button-type-link"
              title="Link"
            >
              <i
                class="rmel-iconfont rmel-icon-link"
              />
            </span>
            <span
              class="button button-type-clear"
              title="Clear"
            >
              <i
                class="rmel-iconfont rmel-icon-delete"
              />
            </span>
            <span
              class="button button-type-undo disabled"
              title="Undo"
            >
              <i
                class="rmel-iconfont rmel-icon-undo"
              />
            </span>
            <span
              class="button button-type-redo disabled"
              title="Redo"
            >
              <i
                class="rmel-iconfont rmel-icon-redo"
              />
            </span>
          </div>
        </div>
        <div
          class="navigation-nav right"
        >
          <div
            class="button-wrap"
          >
            <span
              class="button button-type-mode"
              title="Only display editor"
            >
              <i
                class="rmel-iconfont rmel-icon-keyboard"
              />
            </span>
            <span
              class="button button-type-fullscreen"
              title="Full screen"
            >
              <i
                class="rmel-iconfont rmel-icon-fullscreen"
              />
            </span>
          </div>
        </div>
      </div>
      <div
        class="editor-container"
      >
        <section
          class="section sec-md visible"
        >
          <textarea
            class="section-container input "
            name="contents"
            placeholder="내용을 입력하세요..."
            wrap="hard"
          />
        </section>
        <section
          class="section sec-html visible"
        >
          <div
            class="section-container html-wrap"
          >
            <div
              class="custom-html-style"
            />
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
